### ISSUE TITLE
iput to non-existent resource redirects to default resource without error

### ISSUE THREAD

**Issue by alanking** [irods/irods#4084]
- [x] master
- [x] 4-2-stable

---
### iRODS Version, OS and Version
4.2.3

### What did you try to do?
```
$ iput -R nonExistentResource file
```

### Expected behavior
Failure along with an error indicating that nonExistentResource does not exist.

### Observed behavior (including steps to reproduce, if applicable)
The file will appear in iRODS under the default resource, e.g.:
```
$ ils -l file
/tempZone/home/rods:
  rods              0 demoResc      0 2018-08-28.08:20 & file
```

**Comment by trel** [comment]
This is the defined behavior - the bottom row of the second table at https://docs.irods.org/4.2.3/system_overview/configuration/#default-resource-configuration:


Client | Server | Result | If does_not_exist
-- | -- | -- | --
iput -R 'myResc' foo | msiSetDefaultResc('svr', 'null') | 'myResc' | 'svr'

**Comment by alanking** [comment]
Instead of an error, perhaps we could retain the existing behavior (that is, putting the data object under the default resource) but issue a warning to the user similar to the localhost message. https://github.com/irods/irods/blob/3e6c08a58cfbab723cdf327834c86b4e93bb7eec/scripts/irods/test/test_iadmin.py#L1318

**Comment by trel** [comment]
Agreed - return a warning to the user/client, and put a `LOG_WARNING` into the rodsLog.

**Comment by alanking** [comment]
We have determined that this is not the desired behavior. We want this to be more in line with `irepl` which reports `SYS_RESC_DOES_NOT_EXIST` as seen in this test:

https://github.com/irods/irods/blob/3edd2408610f40d2a6d25a8b8ba6e74baabfaae4/scripts/irods/test/test_irepl.py#L77

**Comment by trel** [comment]
And we'll need to update that table in the Docs.

**Comment by alanking** [comment]
Noting here that this has been fixed in the effort of #4343

**Comment by alanking** [comment]
For posterity: The table in the docs refers to a matrix of situations in which a default resource configured via `msiSetDefaultResc` does not exist. In the case presented by this issue, the default resource exists but the resource targeted by `iput` does not. So there is nothing to update in that particular area of the docs.
